lib/WWW/Mechanize.pm

```diff
@@ -2129,6 +2135,12 @@ sub submit_form {
         $self->{current_form} = $matched[0];
     }
 
+    if (defined($args{strict_forms})) {
```

Comment:
This will turn on `strict_forms` even if `strict_forms => 0` has been passed.  Something like
`if ( $args{strict_forms} ) {` would work.